### PR TITLE
circus: fix incompatible dependency of python-circus

### PR DESCRIPTION
**Pull request by tekeri**
###### Motivation for this change
Fix runtime failure

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @FRidh @volth @etu @xeji

### CODE REVIEW

**Comment by FRidh** [comment]
In the Python packages set only one version of a package at the time is allowed.

**Comment by tekeri** [comment]
OK, how can I fix this package?

**Comment by FRidh** [comment]
Well, aside from introducing tornado 5 compatibility there isn't anything you can do inside `python-packages.nix`.

**Comment by FRidh** [comment]
If it is used *only* as an application, then it could be moved out of `python-packages.nix` and `python-modules/` and use `tornado_4`.

**Comment by tekeri** [comment]
It seems the upstream is not willing to introduce tornado 5 compatibility... ( https://github.com/circus-tent/circus/issues/1064 )
As it is an application, moving it out of python packages set should be the way to go.
Thanks!

File: pkgs/tools/networking/circus/default.nix [hunk 1]
```diff
@@ -1,7 +1,7 @@
-{ stdenv, buildPythonPackage, fetchPypi
```
Comment (FRidh) [inline pkgs/tools/networking/circus/default.nix]: pass in python3 instead of the python packages

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -2309,6 +2309,8 @@ in
     buildGoModule = buildGo112Module;
   };
 
+  circus = python3Packages.callPackage ../tools/networking/circus { };
```
Comment (FRidh) [inline pkgs/top-level/all-packages.nix]: drop the `python3Packages.`

Comment (tekeri) [reply pkgs/top-level/all-packages.nix]: without it `called without required argument 'buildPythonApplication'` appears

File: pkgs/tools/networking/circus/default.nix (hunk 1 above)
Comment (tekeri) [reply pkgs/tools/networking/circus/default.nix]: trying to fix

Comment (tekeri) [reply pkgs/tools/networking/circus/default.nix]: you mean add `python3Packages` to args and use referrence to `python3Packages.buildPythonPackage`?

Comment (tekeri) [reply pkgs/tools/networking/circus/default.nix]: maybe done

**Comment by tekeri** [comment]
Thanks @FRidh

**Pull request merged**